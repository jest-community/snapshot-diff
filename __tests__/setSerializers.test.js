// @flow
const React = require('react');
const { configure, shallow: enzymeShallow } = require('enzyme');
const ReactShallowRenderer = require('react-test-renderer/shallow');
const Adapter = require('enzyme-adapter-react-16');
const enzymeToJson = require('enzyme-to-json/serializer');
const snapshotDiff = require('../src/index');

configure({ adapter: new Adapter() });
const reactShallow = new ReactShallowRenderer();

snapshotDiff.setSerializers([...snapshotDiff.defaultSerializers, enzymeToJson]);

type Props = {
  test: string,
};

const Component = ({ value }) => <div>I have value {value}</div>;

const NestedComponent = (props: Props) => (
  <div>
    <span>Hello World - {props.test}</span>
    <Component value={1234} />
  </div>
);

describe('default rendered components', () => {
  test('diffs components', () => {
    expect(
      snapshotDiff(
        <NestedComponent test="say" />,
        <NestedComponent test="my name" />
      )
    ).toMatchSnapshot();
  });

  test('can use contextLines', () => {
    expect(
      snapshotDiff(
        <NestedComponent test="say" />,
        <NestedComponent test="my name" />,
        {
          contextLines: 0,
        }
      )
    ).toMatchSnapshot();
  });
});

describe('enzyme shallow rendered components', () => {
  test('diffs components', () => {
    expect(
      snapshotDiff(
        enzymeShallow(<NestedComponent test="say" />),
        enzymeShallow(<NestedComponent test="my name" />)
      )
    ).toMatchSnapshot();
  });

  test('can use contextLines', () => {
    expect(
      snapshotDiff(
        enzymeShallow(<NestedComponent test="say" />),
        enzymeShallow(<NestedComponent test="my name" />),
        {
          contextLines: 0,
        }
      )
    ).toMatchSnapshot();
  });
});

describe('react test-renderer shallow rendered components', () => {
  test('diffs components', () => {
    expect(
      snapshotDiff(
        reactShallow.render(<NestedComponent test="say" />),
        reactShallow.render(<NestedComponent test="my name" />)
      )
    ).toMatchSnapshot();
  });

  test('can use contextLines', () => {
    expect(
      snapshotDiff(
        reactShallow.render(<NestedComponent test="say" />),
        reactShallow.render(<NestedComponent test="my name" />),
        {
          contextLines: 0,
        }
      )
    ).toMatchSnapshot();
  });
});

describe('values which are not components', () => {
  test('diffs objects', () => {
    expect(
      snapshotDiff(
        { foo: 'bar', hello: 'world', testing: 123 },
        { foo: 'bar', hello: 'there', testing: 123 }
      )
    ).toMatchSnapshot();
  });

  test('can use contextLines', () => {
    expect(
      snapshotDiff(
        { foo: 'bar', hello: 'world', testing: 123 },
        { foo: 'bar', hello: 'there', testing: 123 },
        {
          contextLines: 0,
        }
      )
    ).toMatchSnapshot();
  });
});
