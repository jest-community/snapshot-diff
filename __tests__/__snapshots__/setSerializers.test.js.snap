// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default rendered components can use contextLines 1`] = `
"Snapshot Diff:
- <NestedComponent test=\\"say\\" />
+ <NestedComponent test=\\"my name\\" />

@@ -4,1 +4,1 @@
-     say
+     my name"
`;

exports[`default rendered components diffs components 1`] = `
"Snapshot Diff:
- <NestedComponent test=\\"say\\" />
+ <NestedComponent test=\\"my name\\" />

@@ -1,9 +1,9 @@
  <div>
    <span>
      Hello World - 
-     say
+     my name
    </span>
    <div>
      I have value 
      1234
    </div>"
`;

exports[`enzyme shallow rendered components can use contextLines 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -4,1 +4,1 @@
-     say
+     my name"
`;

exports[`enzyme shallow rendered components diffs components 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <div>
    <span>
      Hello World - 
-     say
+     my name
    </span>
    <Component
      value={1234}
    />
  </div>"
`;

exports[`react test-renderer shallow rendered components can use contextLines 1`] = `
"Snapshot Diff:
- <div><span>Hello World - say</span><Component value={1234} /></div>
+ <div><span>Hello World - my name</span><Component value={1234} /></div>

@@ -4,1 +4,1 @@
-     say
+     my name"
`;

exports[`react test-renderer shallow rendered components diffs components 1`] = `
"Snapshot Diff:
- <div><span>Hello World - say</span><Component value={1234} /></div>
+ <div><span>Hello World - my name</span><Component value={1234} /></div>

@@ -1,9 +1,9 @@
  <div>
    <span>
      Hello World - 
-     say
+     my name
    </span>
    <div>
      I have value 
      1234
    </div>"
`;

exports[`values which are not components can use contextLines 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -3,1 +3,1 @@
-   \\"hello\\": \\"world\\",
+   \\"hello\\": \\"there\\","
`;

exports[`values which are not components diffs objects 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
    \\"foo\\": \\"bar\\",
-   \\"hello\\": \\"world\\",
+   \\"hello\\": \\"there\\",
    \\"testing\\": 123,
  }"
`;
